File: app/src/main/java/ru/herobrine1st/e621/api/model/Relationships.java
package ru.herobrine1st.e621.api.model;

import androidx.annotation.Nullable;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

@JsonIgnoreProperties({"has_active_children"})
public class Relationships {
    @Nullable
    public Integer getParentId() {
        return parentId;
    }

    public boolean isHasChildren() {
        return hasChildren;
    }

    public List<Integer> getChildren() {
        return children;
    }

    @Nullable
    Integer parentId;
    boolean hasChildren;
    List<Integer> children;
}

File: app/src/main/java/ru/herobrine1st/e621/api/model/Relationships.kt
package ru.herobrine1st.e621.api.model

import android.os.Parcelable
import androidx.compose.runtime.Immutable
import kotlinx.parcelize.Parcelize

@Parcelize
@Immutable
data class Relationships(
    val parentId: Int,
    val hasChildren: Boolean,
    val hasActiveChildren: Boolean,
    val children: List<Int>
) : Parcelable